src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISAwaitable.cs

```diff
@@ -86,7 +86,15 @@ public void UnsafeOnCompleted(Action continuation)
 
         public void Complete(int hr, int cbBytes)
         {
-            _exception = Marshal.GetExceptionForHR(hr);
+            // If the exception was an aborted read operation, ignore it
+            // as we were the one to invoke it.
+            // E_OPERATIONABORTED == 0x800703e3 == -2147023901
+            // TODO check if this error code can be reported through OnAsyncCompletion other than by calling CancelIo.
```

Comment:
Can it?

Reply:
Gotta take a deep dive into the source code. See you in a few weeks.

Reply:
From what Pan I can tell, this error will only be set when CancelIo occurs. So this should be fine.